<?xml version="1.0"?>

<!--

Copyright (c) 2019 Martim BrandÃ£o, Omer Burak Aladag
This software is provided 'as-is', without any express or implied
warranty. In no event will the authors be held liable for any damages
arising from the use of this software.
Permission is granted to anyone to use this software for any purpose,
including commercial applications, and to alter it and redistribute it
freely, subject to the following restrictions:
1. The origin of this software must not be misrepresented; you must not
  claim that you wrote the original software. If you use this software
  in a product, an acknowledgment in the product documentation would be
  appreciated but is not required.
2. Altered source versions must be plainly marked as such, and must not be
  misrepresented as being the original software.
3. This notice may not be removed or altered from any source distribution.

										 -->

<launch>

  <!-- load and publish fsc map -->
  <node pkg="recast_demos" type="gridmap_fsc_demo_node" name="gridmap_fsc_demo_node" respawn="true" output="screen">
  	<param name="resolution" value="0.05" />
  	<param name="path" value="$(find recast_demos)/data/fsc-oil-rig-map-full-1cm-clean.pcd" />
  </node>

  <!-- elevation map filters -->
  <node pkg="grid_map_demos" type="filters_demo" name="elevation_map_filtering" respawn="true" output="screen">
  	<param name="input_topic" value="/gridmap_fsc_demo_node/map" />
  	<param name="output_topic" value="/gridmap_fsc_demo_node/map_filtered" />
    <rosparam command="load" file="$(find recast_demos)/config/filters_demo_filter_chain_res5cm.yaml" />
  </node>

  <!-- save filtered map to file -->
  <node pkg="recast_demos" type="gridmap_save_node" name="gridmap_save_node" respawn="true" output="screen">
  	<param name="input_topic" value="/gridmap_fsc_demo_node/map_filtered" />
  	<param name="save_path" value="$(find recast_demos)/data/gridmap_fsc_res5cm.obj" />
	<param name="save_path_rotated" value="$(find recast_demos)/data/gridmap_fsc_res5cm_rotated.obj" />
  	<param name="save_path_areas" value="$(find recast_demos)/data/gridmap_fsc_res5cm.dat" />
  </node>

  <!-- Launch the grid map visualizer -->
  <node pkg="grid_map_visualization" type="grid_map_visualization" name="grid_map_visualization" output="screen" />

  <!-- Launch RViz with the demo configuration -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find grid_map_demos)/rviz/grid_map_demo.rviz" />

</launch>
